python/lsst/daf/butler/sql/singleDatasetQueryBuilder.py

```diff
@@ -242,38 +224,31 @@ def _columns(selectable, names):
         # full set of link names for this DatasetType
         links = list(datasetType.dimensions.links())
 
-        # Starting point for both subqueries below: a join of dataset to
-        # dataset_collection
+        # Starting point is a join of dataset to dataset_collection
         subJoin = datasetTable.join(
             datasetCollectionTable,
             datasetTable.columns.dataset_id == datasetCollectionTable.columns.dataset_id
         )
         subWhere = and_(datasetTable.columns.dataset_type_name == datasetType.name,
                         datasetCollectionTable.columns.collection.in_(collections))
 
-        # CASE clause that transforms collection name to position in the given
-        # list of collections
-        collorder = case([
-            (datasetCollectionTable.columns.collection == coll, pos) for pos, coll in enumerate(collections)
-        ])
-
-        # first GROUP BY sub-query, find minimum `collorder` for each DataId
-        columns = [functions.min(collorder).label("collorder")] + _columns(datasetTable, links)
-        groupSubq = select(columns).select_from(subJoin).where(subWhere)
-        groupSubq = groupSubq.group_by(*links)
-        groupSubq = groupSubq.alias("sub1" + datasetType.name)
-
-        # next combined sub-query
-        columns = [collorder.label("collorder")] + _columns(datasetTable, ["dataset_id"] + links)
-        combined = select(columns).select_from(subJoin).where(subWhere)
-        combined = combined.alias("sub2" + datasetType.name)
-
-        # now join these two
-        joinsOn = [groupSubq.columns.collorder == combined.columns.collorder] + \
-                  [groupSubq.columns[colName] == combined.columns[colName] for colName in links]
-
-        return cls(registry, fromClause=combined.join(groupSubq, and_(*joinsOn)),
-                   datasetType=datasetType, selectableForDataset=combined, addResultColumns=addResultColumns)
+        # CASE clause that transforms collection name to prefixed name, plus
+        # MIN window function, plus SUBSTR
+        nDigits = 3
+        collFmt = f"{{:0{nDigits}d}}{{}}"
```

Comment:
This is a really minor issue, of course, but it looks like the docs have a ":" between the order and the collection name, while the actual code does not.  Seems best to keep them consistent, just for clarity.